pennylane/gradients/hamiltonian_grad.py

```diff
@@ -24,7 +24,7 @@ def hamiltonian_grad(tape, idx):
         tape (qml.tape.QuantumTape): tape with a single Hamiltonian expectation as measurement
         idx (int): index of parameter that we differentiate with respect to
     """
-    op, p_idx = tape.get_operation(idx)
+    op, _, p_idx = tape.get_operation(idx, return_op_index=True)
```

Comment:
I see two versions for this change:
1. We do it like this now, and have to touch this line again in the version that removes the keyword argument.
2. We do not use the keyword argument here, i.e we do not have to touch this file at all. Then we just have to touch this line later when the default is switched to include the `_, ` part.

It's super minimal but I think option 2 is cleaner. Not a blocker I think.

Reply:
Same applies to the changes in `parameter_shift.py`